### PR TITLE
Upgrade to bootstrap 4

### PR DESCRIPTION
**Pull request by geekygrappler**
supersedes https://github.com/simplabs/ember-simple-auth/pull/1579

### CODE REVIEW

File: tests/dummy/app/templates/components/main-navigation.hbs [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
+    {{#link-to "protected" tagName="li" classNames="nav-item"}}
+      <a class="nav-link">Protected Page</a>
+    {{/link-to}}
+    {{#link-to "auth-error" tagName="li" classNames="nav-item"}}
+      <a class="nav-link">Auth Error Page</a>
+    {{/link-to}}
+  </ul>
+  {{! display logout button when the session is authenticated, login button otherwise }}
+  {{#if session.isAuthenticated}}
+    <a {{action "logout"}} class="btn btn-danger navbar-btn navbar-right">Logout</a>
```
Comment (jessica-jordan) [inline tests/dummy/app/templates/components/main-navigation.hbs]: I think this one could use the `.text-white` color utility class to make the button more readable and to not regress visually from the current layout:

|this branch|master|
|-|-|
|screen shot 2018-04-18 at 14 49 39|screen shot 2018-04-18 at 14 49 46|

See also the color utility class doc here: https://getbootstrap.com/docs/4.1/utilities/colors/#color

**Comment by marcoow** [comment]
Argh, I think `ember-bootstrap` 1.2 doesn't support Ember 1.12 anymore and since we do we can't upgrade…

**Comment by geekygrappler** [comment]
I had seen this was failing, but didn't get a chance to see why. So I'll roll back `ember-bootstrap`?

**Comment by marcoow** [comment]
yes, sadly we have to 😞

**Comment by jessica-jordan** [comment]
This looks good locally, I think we could also add color utility classes to [the login button on the `/login` form as well](https://github.com/simplabs/ember-simple-auth/pull/1585#pullrequestreview-113202608)

> I think adding some color / border to the login button would also be useful to still make it recognizable as a form element:

screen shot 2018-05-04 at 17 59 13

**Pull request closed without merging**